core/rawdb/table.go

```diff
@@ -101,6 +101,12 @@ func (t *table) Sync() error {
 	return t.db.Sync()
 }
 
+// MigrateTable processes the entries in a given table in sequence
+// converting them to a new format if they're of an old format.
+func (t *table) MigrateTable(kind string, isLegacy isLegacyFn, convert convertLegacyFn) error {
+	return t.db.MigrateTable(kind, isLegacy, convert)
```

Comment:
Not sure this `isLegacy` function is needed. Once we decide to migrate/upgrade the freezer table, we just load the legacy value, encode them with latest spec and then store. It's unnecessary to double check if the blob is legacy encoded since the check itself also involves lots of RLP decoding work.

I would suggest just keep this convert function and apply it to all items in the freezer table.

Reply:
With this change the migration went down from 1h30m to 1h16m in my example db. And the code is simpler. Nice suggestion!